pkg/languagedetection/cli_detector.go

```diff
@@ -65,11 +69,52 @@ func languageNameFromCommandLine(cmdline []string) languagemodels.LanguageName {
 }
 
 // DetectLanguage uses a combination of commandline parsing and binary analysis to detect a process' language
-func DetectLanguage(procs []*procutil.Process) []*languagemodels.Language {
+func DetectLanguage(procs []*procutil.Process, sysprobeConfig config.ConfigReader) []*languagemodels.Language {
+	log.Trace("[language detection] Running language detection")
 	langs := make([]*languagemodels.Language, len(procs))
+	unknownPids := make([]int32, 0, len(procs))
+	langsToModify := make(map[int32]*languagemodels.Language, len(procs))
 	for i, proc := range procs {
-		languageName := languageNameFromCommandLine(proc.Cmdline)
-		langs[i] = &languagemodels.Language{Name: languageName}
+		lang := &languagemodels.Language{Name: languageNameFromCommandLine(proc.Cmdline)}
+		langs[i] = lang
+		if lang.Name == languagemodels.Unknown {
+			unknownPids = append(unknownPids, proc.Pid)
+			langsToModify[proc.Pid] = lang
+		}
+	}
+
+	if privilegedLanguageDetectionEnabled(sysprobeConfig) {
+		log.Trace("[language detection] Requesting language from system probe")
+		util, err := net.GetRemoteSystemProbeUtil(
+			sysprobeConfig.GetString("system_probe_config.sysprobe_socket"),
+		)
+		if err != nil {
+			log.Warn("[language detection] Failed to request language:", err)
+			return langs
+		}
+
+		privilegedLangs, err := util.DetectLanguage(unknownPids)
```

Comment:
Doesn't necessarily have to be in this PR, but since we've historically seen performance issues in process agent -> system probe communication paths, should we add some additional telemetry around this? E.g. logging how long the language detection took every 20 checks, similar to the process check log output

Reply:
Added histogram telemetry around this.